pkg/reconciler/status.go

```diff
@@ -116,9 +122,21 @@ func (s *Status) SetReconcileCompleted(err error) {
 	s.UpdateFunc(s.S)
 }
 
-func (s *Status) SetDeleteCompleted(err error) {
-	s.removeAllConditions()
+// SetPackageYanked appends a new condition to indicate a package is yanked
+func (s *Status) SetPackageYanked(meta metav1.ObjectMeta, reason string) {
+	s.markObservedLatest(meta)
```

Comment:
dont think we should mark latest generation here since this is not part of primary reconciliation logic (e.g. Reconcile*)